lgc/patch/PatchInOutImportExport.cpp

```diff
@@ -5436,6 +5449,65 @@ Value *PatchInOutImportExport::reconfigWorkgroup(Value *localInvocationId, Instr
   return remappedId;
 }
 
+// =====================================================================================================================
+// This function adds instructions to swizzle thread ID inside a group.
+//
+// The data layout in TCP is typically 8 x something.
+// The typewriter style SIMD layout does not match well with the data layout,
+// which will potentially cause partial cache line read/write.
+//
+// In the case of 16x16 thread group size, wf32 SIMD will be created in eight 16x2 regions.
+// This function maps the eight 16x2 regions into eight 8x4 regions.
+//
+// Originally SIMDs cover 16x2 regions, after swizzling they cover 8x4 regions.
+//
+// @param localInvocationId : The original workgroup ID.
+// @param insertPos : Where to insert instructions.
+Value *PatchInOutImportExport::swizzleLocalInvocationIdIn8x4(Value *localInvocationId, Instruction *insertPos) {
+  IRBuilder<> builder(*m_context);
+  builder.SetInsertPoint(insertPos);
+  auto &mode = m_pipelineState->getShaderModes()->getComputeShaderMode();
+
+  const unsigned workgroupSizeX = mode.workgroupSizeX;
+  const unsigned workgroupSizeY = mode.workgroupSizeY;
+
+  if ((workgroupSizeX >= 16) && (workgroupSizeX % 8 == 0) && (workgroupSizeY % 4 == 0)) {
+    Value *threadIdInGroupX = builder.CreateExtractElement(localInvocationId, (uint64_t)0);
+    Value *threadIdInGroupY = builder.CreateExtractElement(localInvocationId, 1);
+    Value *threadIdInGroupZ = builder.CreateExtractElement(localInvocationId, 2);
+    const unsigned blockSizeX = 8;
+    const unsigned blockSizeY = 4;
+    const unsigned blockDimY = workgroupSizeY / 4; // How many blocks are splitted in Y level.
+
+    unsigned blockSize = blockSizeX * blockSizeY; // 32 threads per-block
```

Comment:
Since other variables are `const`, can we mark `blockSize` as well? I don't see any reassignments below.